### ISSUE TITLE
Assembly function definition not working

### ISSUE THREAD

**Issue by PaulRBerg** [duaraghav8/Ethlint#248]
Suppose we have the following syntax:

```solidity
pragma solidity ^0.4.24;

contract Example {
    function() external payable {
        assembly {
          function myAwesomeFunc() {
            let foo := "bar"
          }
        }
    }
}
```

I get the following error when linting with solium, despite the fact truffle successfully compiles the code:

> 6:11 Syntax error: unexpected token f

**Comment by duaraghav8** [comment]
Thanks for reporting this @PaulRBerg.
The fix for this will be added in the next release

**Comment by duaraghav8** [comment]
Resolved in `v1.2.2`.
